tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/PublicClientApplicationBuilderTests.cs

```diff
@@ -103,6 +103,48 @@ public void TestConstructor_WithDebugLoggingCallback()
             Assert.IsTrue(pca.AppConfig.IsDefaultPlatformLoggingEnabled);
         }
 
+        [TestMethod]
+        public void TestConstructor_WithDebugLoggingCallbackAndAppConfig()
+        {
+            // Ensure that values in the options are not reset to defaults when not sent into WithLoggingCallback
+            var options = new PublicClientApplicationOptions
+            {
+                ClientId = MsalTestConstants.ClientId,
+                LogLevel = LogLevel.Error,
+                EnablePiiLogging = true,
+                IsDefaultPlatformLoggingEnabled = true
+            };
+
+            var pca = PublicClientApplicationBuilder.CreateWithApplicationOptions(options)
+                .WithLoggingCallback((level, message, pii) => { }).Build();
+
+            Assert.AreEqual(LogLevel.Error, pca.AppConfig.LogLevel);
+            Assert.IsTrue(pca.AppConfig.EnablePiiLogging);
+            Assert.IsTrue(pca.AppConfig.IsDefaultPlatformLoggingEnabled);
+        }
+
+        [TestMethod]
+        public void TestConstructor_WithDebugLoggingCallbackAndAppConfigWithOverride()
+        {
+            // Ensure that values in the options are reset to new values when sent into WithLoggingCallback
+            var options = new PublicClientApplicationOptions
+            {
+                ClientId = MsalTestConstants.ClientId,
+                LogLevel = LogLevel.Error,
+                EnablePiiLogging = false,
+                IsDefaultPlatformLoggingEnabled = true
+            };
+
+            var pca = PublicClientApplicationBuilder.CreateWithApplicationOptions(options)
+                .WithLoggingCallback((level, message, pii) => { },
```

Comment:
This method is a bit strange - it's called WithLoggingcallback, but it actually accepts 4 params, the first being a LogCallback. Maybe rename it as WithLogging ?

Reply:
WithLogging sounds good, will change.

Reply:
Also changing WithTelemetryCallback to WithTelemetry for consistency.